rma/models/rma_order_line.py

```diff
@@ -597,3 +618,26 @@ def action_view_procurements(self):
             result['views'] = [(res and res.id or False, 'form')]
             result['res_id'] = procurements[0]
         return result
+
+    @api.multi
+    def action_view_rma_lines(self):
+        if self.type == 'customer':
+            # from customer we link to supplier rma
+            action = self.env.ref(
+                'rma.action_rma_supplier_lines')
+            rma_lines = self.supplier_rma_line_ids.ids
+            res = self.env.ref('rma.view_rma_line_supplier_form', False)
+        else:
+            # from supplier we link to customer rma
+            action = self.env.ref(
+                'rma.action_rma_customer_lines')
+            rma_lines = self.customer_rma_id.ids
+            res = self.env.ref('rma.view_rma_line_form', False)
+        result = action.read()[0]
+        # choose the view_mode accordingly
+        if len(rma_lines) != 1:
```

Comment:
it fails if the supplier rma lines is not created yet

Reply:
yes, that would be great

Reply:
done, but there was in fact a bug in the method. Fixed.